src/components/FilterList/SearchSettingsMenu/FilterListSearchSettingsMenu.tsx

```diff
@@ -49,10 +42,10 @@ export function FilterListSearchSettingsMenu<T extends object>({
           {/* When using Object.entries a new object is created, and we cant infer the type from that */}
           {Object.entries(keys).map(([key, enabled]) => (
             <Group key={key}>
-              <Checkbox
-                label={labels ? labels[key as keyof T] : camelCaseToRegularText(key)}
-                checked={enabled as boolean}
-                onChange={(event) => setKey(key as keyof T, event.currentTarget.checked)}
+              <BoolInput
+                name={labels ? labels[key as keyof T] || '' : camelCaseToRegularText(key)}
```

Comment:
I think in the case that we have a labels object but that we don't have that particular key in the object, we want to fall back on the camelCaseToRegularText, not an empty string.

Example: 
```
keys = {
action: true,
guiPath: false
}

labels = {
guiPath: "User Interface Path"
}
```
In this case, action will become an empty string in this line.
Alternatively, we can enforce in the types that all the keys in `keys` must also be defined in the `labels` prop, if `labels`is passed

Reply:
Let me see if I understand this correctly. This should work and address what you're mentioning here, right?
```
                label={(labels && labels[key as keyof T]) || camelCaseToRegularText(key)}
```

Reply:
Correct, you can also write it like this

```
labels?.[key as keyof T] || camelCaseToRegularText(key)
```